src/layers/perfetto.rs

```diff
@@ -90,10 +92,23 @@ impl Layer {
     /// - `PERFETTO_BIN_PATH`: path to the perfetto binaries. If not set, the system path will be used. Is used only with the system backend.
     /// - `PERFETTO_CFG_PATH`: path to the perfetto config file. If not set, the default one `config/system_profiling.cfg` will be used. Is used only with the system backend.
     /// - `PERFETTO_BUFFER_SIZE_KB`: size of the buffer in kilobytes. Default: 50 * 1024. Is used only with the in-process backend.
+    /// - `PERFETTO_PLATFORM_NAME`: custom platform name. Default: architecture of the CPU that is currently in use.
     pub fn new_from_env() -> Result<(Self, PerfettoGuard), perfetto_sys::Error> {
         let trace_file_patch = match std::env::var("PERFETTO_TRACE_FILE_PATH") {
             Ok(path) => path,
-            Err(_) => "tracing.perfetto-trace".to_string(),
+            Err(_) => {
+                let timestamp = get_unix_timestamp();
+                let branch = get_current_git_branch();
+                let platform = std::env::var("PERFETTO_PLATFORM_NAME")
+                    .unwrap_or(std::env::consts::ARCH.to_string());
+
+                format!(
+                    "{}-{}{}.perfetto-trace",
+                    timestamp,
+                    platform,
+                    branch.map_or(String::new(), |b| format!("-{}", b))
```

Comment:
IMO better change the order as we will usually compare the results for the same machine and different branches, i.e. platform-branch-time